edge-hub/test/Microsoft.Azure.Devices.Routing.Core.Test/endpoints/StoringAsyncEndpointExecutorTest.cs

```diff
@@ -288,6 +295,128 @@ public async Task StoreMessagesProviderIntermittantMessagesTest()
+            Assert.Contains(normalPriMsg4, normalPriQueue);
+
+            var lowPriQueue = messageStore.GetReceivedMessagesForEndpoint($"{endpoint.Id}_Pri{LowPri}");
+            Assert.Equal(2, lowPriQueue.Count);
+            Assert.Contains(lowPriMsg1, lowPriQueue);
+            Assert.Contains(lowPriMsg2, lowPriQueue);
+
+            // Re-enable the endpoint and let the queues drain
+            endpoint.CanProcess = true;
+            await Task.Delay(TimeSpan.FromSeconds(10));
+
+            // Assert - Make sure the endpoint received all the messages
+            // in the right priority order:
+            //  - HighPri messages should finish processing before others
+            //  - NormalPri messages should finish processing before LowPri
+            Assert.Equal(8, endpoint.Processed.Count());
+            int highPriMessagesProcessed = 0;
+            int normalPriMessagesProcessed = 0;
+            int lowPriMessagesProcessed = 0;
+            for (int i = 0; i < endpoint.Processed.Count(); i++)
+            {
+                IMessage message = endpoint.Processed[i];
+                if (message.Properties.ContainsKey($"highPriority"))
+                {
+                    if (++highPriMessagesProcessed == HighPriCount)
+                    {
+                        // Found all the high-pri messages,
+                        // normal and low pri at this point
+                        // must not have completed yet
+                        Assert.True(normalPriMessagesProcessed < NormalPriCount);
+                        Assert.True(lowPriMessagesProcessed < LowPriCount);
+                    }
+                }
+                else if (message.Properties.ContainsKey($"normalPriority"))
+                {
+                    if (++normalPriMessagesProcessed == NormalPriCount)
+                    {
+                        // Found all the normal-pri messages,
+                        // low pri messages at this point must
+                        // not have completed yet
+                        Assert.True(lowPriMessagesProcessed < LowPriCount);
+
+                        // High pri messages should have completed
+                        Assert.True(highPriMessagesProcessed == HighPriCount);
+                    }
+                }
+                else if (message.Properties.ContainsKey($"lowPriority"))
+                {
+                    if (++lowPriMessagesProcessed == LowPriCount)
+                    {
+                        // Found all the low-pri messages,
+                        // high-pri and normal-pri should also
+                        // have completed before this
+                        Assert.True(highPriMessagesProcessed == HighPriCount);
+                        Assert.True(normalPriMessagesProcessed == NormalPriCount);
+                    }
+                }
+                else
+                {
+                    // Bad test setup
+                    Assert.True(false);

```

Comment:
nit: suggest to use Assert.True(false, "Bad test setup or even tell how to resolve it.") with description argument.